### PR TITLE
Fix stats reset bug when websocket worker nodes rejoin.

### PR DESCRIPTION
**Pull request by Bo887**
Addresses #17.

Checks to see if `self.worker_node` exists before creating a new one.

### CODE REVIEW

**Comment by ayushr2** [comment]
Thanks for looking into this Eric! So the scenario we want to handle is: when a worker node disconnects (and its corresponding `WorkerConnectionHandler` object is deleted because the connection broke. But it exists in the DB still. Then when it reconnects again, we want to sustain the worker node object from the DB and just flip the `worker_node.is_alive` boolean to true.

In your approach you check if `self.worker_node` is `None` or not. Which will not work because we do not sustain this object across connections. You will need to see if the worker had connected before or not by querying the DB. We already attempt to do that in https://github.com/illinois-cs241/broadway/blob/master/broadway/api/handlers/worker_ws.py#L54-L77. But our implementation is buggy.

File: broadway/api/handlers/worker_ws.py [hunk 1]
```diff
@@ -41,22 +41,25 @@ def handler_register(self, hostname):
...
+            )
             logger.info(
                 "new worker '{}' joined on '{}'".format(self.worker_id, hostname)
             )
             worker_node_dao.insert(self.worker_node)
         elif not dup.is_alive:
+            self.worker_node = dup
+            self.worker_node.hostname = hostname
+            self.worker_node.last_seen = get_time()
+            self.worker_node.is_alive = True
```
Comment (ayushr2) [inline broadway/api/handlers/worker_ws.py]: Can we also add `self.use_ws =  True`? In a very very unusual case, we could by mistake connect an "old" HTTP worker with an ID. then shut it down and then connect it with the same ID with websockets. Then we would need to update this field.

**Comment by Bo887** [comment]
Should I merge this now then?

**Comment by ayushr2** [comment]
Yes go ahead!

**Comment by ayushr2** [comment]
A cool feature about Github is that you could write "Resolves #<issue number>" in your description and it will automatically close the issue when the PR is merged. So just replacing "Addresses" with "Resolves" in your description would have done the job for this PR.

**Pull request merged**